.dev/script/new-traffic-data_20210508.R

```diff
@@ -0,0 +1,228 @@
+      Second_str, # Second street name, need to be joined with A1_street_name
+      Overtaking, # E.g. One vehicle, two or more, etc.
+      Within_70m, # Whether within 70m of junction
+      Road_type_, # E.g. One way, Two way, Dual carriageway
+      Pedal_cycl, # Others, Pedal Cycle, Motorcycle
+      TypeOfCo_P, # E.g. Vehicle collision with Pedestrian, etc.
+      Struc_Type, # E.g. At grade road, Flyover, Bridge, etc.
+      RD_Class_L, # E.g. Expressway, Main Road, Secondary Road, etc.
+      Road_class, # Slightly different from `Road_Classification`
+      
+      # New variables that do not require code frame -----------------
+      Precise_lo, # Precise location. Strength.
+      Accident_a, # No/ Yes
+      Junction_t, # E.g. Cross roads, T-junction, Y-junction, Multiple, etc.
+      Whether_at, # E.g. No crossing control, On a crossing control, etc.
+      Type_of_Cr, # E.g. Footbridge/Subway, Traffic signal, Zebra, etc.
+      
+      ),
+    by = c("Date", "Serial_No_", "Year")
+  )
+  
+  # labelling ----------------------------------------------------------------
+
+hk_accidents_new_cleaned_labelled <-
+  hk_accidents_new_cleaned_unlabelled %>%
+  mutate(Street_Name = look_up(Street_nam, dictionary = t_A1_street_name)) %>%
+  mutate(Second_Street_Name = look_up(Second_str, dictionary = t_A1_street_name)) %>%
+  mutate(Overtaking = look_up(Overtaking, dictionary = t_Overtaking)) %>%
+  mutate(Within_70m = look_up(Within_70m, dictionary =  t_Within_70m)) %>%
+  mutate(Road_Type = look_up(Road_type_, dictionary = t_Road_type_)) %>%
+  mutate(Cycle_Type = look_up(Pedal_cycl, dictionary = t_Ped_Cycle)) %>%
+  mutate(Type_of_Collision_v2 = look_up(TypeOfCo_P, dictionary = t_Collision_Type)) %>%
+  mutate(Structure_Type = look_up(Struc_Type, dictionary = t_Struc_Type)) %>%
+  mutate(Road_Class_L = look_up(RD_Class_L, dictionary = t_Road_class_L)) %>%
+  mutate(Road_Classification_v2 = look_up(Road_class, dictionary = t_Road_class)) %>%
+  
+  # Remove old names ------------------------------------------------------
+  select( 
+    -Street_nam,
+    -Second_str,
+    -Road_type_,
+    -Pedal_cycl,
+    -TypeOfCo_P,
+    -Struc_Type,
+    -RD_Class_L,
+    -Road_class
+  ) %>%
+  
+  # Rename new variables --------------------------------------------------
+  rename(Precise_Location = "Precise_lo") %>%
+  rename(Accident = "Accident_a") %>%
+  rename(Junction_Type = "Junction_t") %>%
+  rename(Crossing_Control = "Whether_at") %>%
+  rename(Crossing_Type = "Type_of_Cr") %>%
+  rename(Type_of_Collision_with_cycle = "Type_of_Collision_v2") %>%
+  rename(Road_Hierarchy = "Road_Class_L") %>%
+
+  # Rename rows with value "n.a." to "NA" ---------------------------------
+  mutate(Junction_Type = case_when(Junction_Type == "n.a." ~ "NA",
+                                   TRUE ~ Junction_Type)) %>%
+  mutate(Crossing_Type = case_when(Crossing_Type == "n.a." ~ "NA",
+                                 TRUE ~ Junction_Type)) %>%
+  
```

Comment:
I found **Road_Hierarchy** also have NA values, in the format of **N.A.** though.

```r
> unique(hkdatasets::hk_accidents$Road_Hierarchy)
[1] "Expressway"       "Main Road"        "Secondary Road"   "Other Minor Road" "N.A."   
```

```suggestion
  mutate(Road_Hierarchy = ifelse(Road_Hierarchy == "N.A.", NA, Road_Hierarchy))
```